test/services/user-group/services/courses.test.js

```diff
@@ -0,0 +1,80 @@
+const { expect } = require('chai');
+const app = require('../../../../src/app');
+const testObjects = require('../../helpers/testObjects')(app);
+const { courseModel } = require('../../../../src/services/user-group/model');
+
+describe('course service', () => {
+	let server;
+
+	before((done) => {
+		server = app.listen(0, done);
+	});
+
+	after(async () => {
+		await testObjects.cleanup();
+		await server.close();
+	});
+
+
+	it('CREATE a course', async () => {
+		const { _id: schoolId } = await testObjects.createTestSchool({});
+		const teacher = await testObjects.createTestUser({ roles: ['teacher'], schoolId });
+		const params = await testObjects.generateRequestParamsFromUser(teacher);
+		const result = await app.service('courses').create({
+			name: 'testcourse',
+			schoolId: schoolId.toString(),
+			teacherIds: [teacher._id],
+			substitutionIds: [],
+			classIds: [],
+			userIds: [],
+		}, params);
+		expect(result).to.not.be.undefined;
+		expect(result).to.haveOwnProperty('_id');
+		expect(result.name).to.eq('testcourse');
+		expect(result.schoolId.toString()).to.eq(schoolId.toString());
+		// make sure course was saved to db
+		const dbResult = await courseModel.findById(result._id).lean().exec();
+		expect(dbResult).to.not.be.undefined;
+		expect(dbResult).to.haveOwnProperty('_id');
+		expect(dbResult.name).to.eq('testcourse');
+		expect(dbResult.schoolId.toString()).to.eq(schoolId.toString());
+	});
+
+	it('GET a course', async () => {
+		const { _id: schoolId } = await testObjects.createTestSchool({});
+		const teacher = await testObjects.createTestUser({ roles: ['teacher'], schoolId });
+		const course = await testObjects.createTestCourse({ schoolId, teacherIds: [teacher._id] });
+		const params = await testObjects.generateRequestParamsFromUser(teacher);
+		const result = await app.service('courses').get(course._id, params);
+		expect(result).to.not.be.undefined;
```

Comment:
Chai deep equal für expect(course).to.deep.equal(course) oder .to.have.all.deep.keys könnte die Test stabiler machen. Habe aber beides auch noch nicht ausprobiert. Ob das für .toString() === .toString() funktioniert, müsste man aber ausprobieren.
https://www.chaijs.com/api/bdd/#method_deep

Reply:
habe ich ausprobiert, wirft fehler weil das Object nach dem create anders ist als das ursprünglich reingegebene (schon allein dass da ne _id dazukommt und so...)

Reply:
Ach ok, ist ein Argument, daran habe ich nicht gedacht.